libsolidity/interface/CompilerStack.h

```diff
@@ -266,7 +276,7 @@ class CompilerStack: boost::noncopyable
 	/// list of path prefix remappings, e.g. mylibrary: github.com/ethereum = /usr/local/ethereum
 	/// "context:prefix=target"
 	std::vector<Remapping> m_remappings;
-	bool m_parseSuccessful;
+	bool m_success;
```

Comment:
This should probably be more fine-grained here. Currently, it is possible to call `analyze()` even after an unsuccessful parse. I think we should introduce an enum that represents the compiler stack state:
`SourcesSet`, `ParsingSuccessful`, `AnalysisSuccessful`, `CompilationSuccessful`
With appropriate checks at the start of each public functio.